### ISSUE TITLE
Location issues

### ISSUE THREAD

**Issue by ichiriac** [glayzzle/php-parser#164]
Related to https://github.com/prettier/plugin-php/issues/246 :

```php
<?php

// good behaviour
declare (foo = true);
echo $foo;
return $foo;
continue;
break;
global $foo, $bar;
static $foo;
class foo {}
throw new \Exception($bad);

// bad behaviour
const foo = 1;
static::$foo;
foo::bar;
class ok {
  static function bad() { }
}
// any expr based statement
$foo = 1 + 2;
```

**Comment by ichiriac** [comment]
This fix is now relased under 3.0.0-alpha.3

**Comment by ichiriac** [comment]
3.0.0-prerelease.4
